chore(http): Refresh some headers pages (~a-c) (#36038)
mdn/content: files/en-us/web/http/headers/access-control-allow-credentials/index.md

{{HTTPSidebar}}

The HTTP **`Access-Control-Allow-Credentials`** response header tells browsers whether the server allows credentials to be included in cross-origin HTTP requests.

> Credentials include cookies, {{glossary("TLS")}} client certificates, or authentication headers containing a username and password.
> By default, these credentials are not sent in cross-origin requests, and doing so can make a site vulnerable to {{Glossary("CSRF")}} attacks.

Reviewer: let's also expand these terms here:
```suggestion
Credentials include cookies, {{glossary("TLS", "Transport Layer Security (TLS)")}} client certificates, or authentication headers containing a username and password.
By default, these credentials are not sent in cross-origin requests, and doing so can make a site vulnerable to {{Glossary("CSRF", "Cross-Site Request Forgery (CSRF)")}} attacks.
```

Author: nice

> A client can ask that credentials should be included in cross-site requests in several ways:

Reviewer: ```suggestion
A client can ask credentials to be included in cross-site requests in several ways:
```

> If the client has asked for credentials to be included:

Reviewer: ```suggestion
When credentials are included:
```

> - If the request is {{glossary("Preflight_request", "preflighted")}}, then the preflight request does not include credentials.
>   If the server's response to the preflight request sets the `Access-Control-Allow-Credentials` header to `true`, then the real request will include credentials: otherwise, the browser reports a network error.
> - If the request is not preflighted, then the request will include credentials, and if the server's response does not set the `Access-Control-Allow-Credentials` header to `true`, the browser reports a network error.

Reviewer: we can change the format to:
```suggestion
- For {{glossary("Preflight_request", "preflighted")}} requests: The preflight request does not include credentials.
  If the server's response to the preflight request sets the `Access-Control-Allow-Credentials` header to `true`, then the real request will include credentials; otherwise, the browser reports a network error.
- For non-preflighted requests: The request will include credentials, and if the server's response does not set the `Access-Control-Allow-Credentials` header to `true`, the browser reports a network error.
```

## Directives

- `true`

>   - : The only valid value for this header is `true` (case-sensitive).
>     If you don't need credentials, omit this header entirely rather than setting its value to `false`.

Reviewer: A description is missing
```suggestion
  - : The server allows credentials to be included in cross-origin HTTP requests.
     This is the only valid value for this header and is case-sensitive.
     If you don't need credentials, omit this header entirely rather than setting its value to `false`.
```
